src/stores/chat.ts

```diff
@@ -283,18 +249,28 @@ export const useChatStore = defineStore('chat', () => {
 			const threadIds = Object.keys(threadIdSetsToUpdate)
 			if (threadIds.length) {
 				const chatBlockWithMergeBy: Set<number> | undefined = chatBlocks[token]?.find((set) => set.has(options.mergeBy!))
-				if (chatBlockWithMergeBy) {
+				threadIds.forEach((threadId) => {
+					// Fallback: This is either polling or posting new message in a thread view
+					// with no other messages in main chat block that thread belongs to
+					const lastMessageIdInThread = chatExtrasStore.getThread(token, +threadId)?.last?.id
```

Comment:
by this time, `processMessage` already updates the thread info. We could either:
- make sure we trigger `processChatBlocks` before `processMessage`
- check, if temporary message id can be replaced in chat blocks

Reply:
Please also rebase, it's behind the fix with animation effect